cicerone/app/src/main/java/com/teampower/cicerone/GeofenceTriggeredActivity.kt

```diff
@@ -107,17 +133,59 @@ class GeofenceTriggeredActivity : AppCompatActivity()  {
             }
         })
         tts_button.setOnClickListener {
-            if(speaking){
+            if (speaking) {
                 tts.stop()
                 speaking = false
-            } else{
+                tts_button.setImageResource(R.drawable.ic_play_arrow_black_32dp)
+            } else {
                 tts.speak(tts_text, TextToSpeech.QUEUE_FLUSH, null, "TRIG_TTS")
+                tts_button.setImageResource(R.drawable.ic_stop_black_32dp)
             }
         }
+
+        favorite_button.setOnClickListener { toggleFavoritePOI(POI) }
         Log.v(TRIG_TAG, POI.toString())
 
         // Demo on how to update category score
         catViewModel = ViewModelProvider(this).get(CategoryViewModel::class.java)
         catViewModel.updateCategoryPoints("School", 2.0) // Change score
     }
+
+    private fun toggleFavoritePOI(poi: POI) {
+        MainScope().launch {
+            val result = poiSavedViewModel.loadPOI(poi.id).await()
+            isSaved = result !== null
+
+            if (!isSaved) {
+                // Add to favorites
+                val currentTimeString = Instant.now().toString()
+                poiSavedViewModel.favorite(
+                    POISavedData(
+                        poi.id,
+                        poi.name,
+                        poi.category,
+                        currentTimeString,
+                        poi.lat,
+                        poi.long
+                    )
+                )
```

Comment:
This can just be replaced with placedetails if we extend the schema, if we want to rebuild the GeofenceTriggeredActivity.